core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -1062,6 +1062,11 @@ public String getMessage() {
   public static final String KETTLE_GLOBAL_LOG_VARIABLES_CLEAR_ON_EXPORT = "KETTLE_GLOBAL_LOG_VARIABLES_CLEAR_ON_EXPORT";
 
   /**
+   * Set this variable to true to disable receiving file names from another step for XML Input Stream (StAX) step.
+   */
+  public static final String KETTLE_IGNORE_RECEIVING_INPUT = "KETTLE_IGNORE_RECEIVING_INPUT";
```

Comment:
@mbatchelor, I've created property to make it backward compatible with this issue #826.
Just to be clear regarding "make it behave as it did before" did you mean totally disable influence of incoming hops to the step( only use entered manually variables ) or give a priority to entered manually variables over incoming hops?